lib/ui/src/components/notifications/notifications.tsx

```diff
@@ -28,11 +27,22 @@ const NotificationList: FunctionComponent<{
   placement: CSSObject;
   notifications: State['notifications'];
 }> = ({ notifications, placement = undefined }) => {
+  const [dismissedNotificationsIds, setDismissedNotificationsWithIds] = React.useState<string[]>(
+    []
+  );
   return (
     <List placement={placement}>
-      {notifications.map((notification) => (
-        <NotificationItem key={notification.id} notification={notification} />
-      ))}
+      {notifications.map((notification) =>
+        dismissedNotificationsIds.indexOf(notification.id) === -1 ? (
+          <NotificationItem
+            key={notification.id}
+            setDismissedNotification={(id: string) =>
+              setDismissedNotificationsWithIds([...dismissedNotificationsIds, ...[id]])
+            }
+            notification={notification}
+          />
+        ) : null
+      )}
```

Comment:
What's the reason behind having a list of dismissed notifications id's rather then removing the notification from the state?

https://github.com/storybookjs/storybook/blob/341fc2d90d4f3ae42fd9b68173b9f7b1856917f0/lib/api/src/modules/notifications.ts#L30-L41

Reply:
I must have missed this. I  changed the implementation to use `clearNotification` instead of hiding the notifications.